core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -178,8 +178,15 @@ private function setPredefinedImageSizes(array $config, ContainerBuilder $contai
 
         $imageSizes = [];
 
+        // Do not add a size with the special name '_default' but merge its values into all other definitions instead.
         foreach ($config['image']['sizes'] as $name => $value) {
-            $imageSizes['_'.$name] = $this->camelizeKeys($value);
+            if ('_defaults' === $name) {
+                continue;
+            }
+
+            $imageSizes['_'.$name] = $this->camelizeKeys(
+                array_merge($config['image']['sizes']['_defaults'] ?? [], $value)
```

Comment:
`array_merge()` is an expensive function that should not be called in a foreach loop. I do understand that we have to in this case, but can we at least check if there are any defaults at all and skip the `array_merge()` call if there are not?

Reply:
This only scales with the number of configured image sizes and it's only executed at compile time, so I don't think performance should be a concern.

That said, would you prefer sth. like this?
```php
if(isset($config['image']['sizes']['_defaults'])) {
    $value = array_merge($config['image']['sizes']['_defaults'], $value);
}

$imageSizes['_'.$name] = $this->camelizeKeys($value);
```

Reply:
Yes, I would. 😊

Reply:
Interestingly this is quicker even with set `_defaults`: https://3v4l.org/FihqU. :smile: But I wouldn't bet if this generalizes with real data. And we're talking <1ms for 100 image sizes.

Reply:
Done in b3d9eb08a2e5f26a5b44ef861479b48881826719.